library/bignum.c

```diff
@@ -983,7 +983,7 @@ int mbedtls_mpi_add_mpi( mbedtls_mpi *X, const mbedtls_mpi *A, const mbedtls_mpi
     MPI_VALIDATE_RET( B != NULL );
 
     s = A->s;
-    if( A->s * B->s < 0 )
+    if( A->s ^ B->s )
```

Comment:
That's a fancy way of writing `if( A->s != B->s )`, with the added twist that the combination of bitwise operators and signed integers makes the reader pause and think whether it might create a trap representation. (It won't, because the only allowed values are 1 and -1.)

Reply:
...and how excessive is `A->s * B->s > 0` to do the same?
:rocket: